tony-core/src/main/java/com/linkedin/tony/TonyApplicationMaster.java

```diff
@@ -126,6 +126,8 @@
   private AtomicInteger numRequestedContainers = new AtomicInteger();
   private Map<String, List<ContainerRequest>> jobTypeToContainerRequestsMap = new HashMap<>();
 
+  private long lastAllocationRequestId = 0;
```

Comment:
I don't think we should reset this on retry. (retry here I assume refers to retrying unregistered tasks) IMO there is a separate issue to be addressed: in TonyApplicationMaster#rescheduleTasks we call removeAllContainerRequests which removes all requests in AMRMClient, but rescheduleTasks is called only for the allocated but unregistered tasks. So when we retry allocated but unregistered tasks, there still may be outstanding container requests which have not even been allocated. So we don't want to reset since it will collide with existing ids in TensorFlowSession#jobTypeToAllocationIds

Reply:
No, I meant restarting the whole thing, we will create a new TensorFlowSession in the next retry. However on a second thought, I think this should be kept as is since we don't want to reset containerRequestId for containerRequest.

Reply:
Oh gotcha. Yeah that's a good point.